zebra/zebra_vxlan.c

```diff
@@ -9637,3 +9657,66 @@ static void zebra_vxlan_sg_cleanup(struct hash_backet *backet, void *arg)
 
 	zebra_vxlan_sg_del(vxlan_sg);
 }
+
+/************************** EVPN BGP config management ************************/
+static void zvni_evpn_cfg_cleanup(struct hash_bucket *bucket, void *ctxt)
+{
+        zebra_vni_t *zvni = NULL;
+	zebra_l3vni_t *zl3vni = NULL;
+
+        zvni = (zebra_vni_t *)bucket->data;
+	zvni->advertise_gw_macip = 0;
+	zvni->advertise_svi_macip = 0;
+	zvni->advertise_subnet = 0;
+
+	zl3vni = zl3vni_from_vrf(zvni->vrf_id);
+	if (zl3vni)
+	    listnode_delete(zl3vni->l2vnis, zvni);
+
+	zvni_neigh_del_all(zvni, 0, 0, DEL_ALL_NEIGH);
+	zvni_mac_del_all(zvni, 0, 0, DEL_ALL_MAC);
```

Comment:
neigh_del_all and mac_del_all also cleans up locally learn neighbors and mac addresses. How would they recover when bgpd comes up ?